src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,241 @@
+"""Module for deconvolution of mass transport effects."""
+
+from .ec_ms import ECMSMeasurement
+from scipy.optimize import curve_fit  # noqa
+from scipy.interpolate import interp1d  # noqa
+from scipy import signal  # noqa
+from mpmath import invertlaplace, sinh, cosh, sqrt, exp, erfc, pi, tanh, coth  # noqa
+import matplotlib.pyplot as plt
+from numpy.fft import fft, ifft, ifftshift, fftfreq  # noqa
+import numpy as np
+
+
+class DecoMeasurement(ECMSMeasurement):
+    """Class implementing deconvolution of EC-MS data"""
+
+    def __intit__(self, name, **kwargs):
+        """initialize a deconvolution EC-MS measurement
+
+        Args:
+            name (str): The name of the measurement"""
+        super().__init__(name, **kwargs)
+
+    def grab_partial_current(
+        self, signal_name, kernel_obj, tspan=None, tspan_bg=None, snr=10
+    ):
+        """Return the deconvoluted partial current for a given signal
+
+        Args:
+            signal_name (str): Name of signal for which deconvolution is to
+                be carried out.
+            kernel_obj (Kernel): Kernel object which contains the mass transport
+                parameters
+            tspan (list): Timespan for which the partial current is returned.
+            tspan_bg (list): Timespan that corresponds to the background signal.
+            snr (int): signal-to-noise ratio used for Wiener deconvolution.
+        """
+
+        t_sig, v_sig = self.grab_cal_signal(signal_name, tspan=tspan, tspan_bg=tspan_bg)
+
+        kernel = kernel_obj.calculate_kernel(
+            dt=t_sig[1] - t_sig[0], duration=t_sig[-1] - t_sig[0]
+        )
+        kernel = np.hstack((kernel, np.zeros(len(v_sig) - len(kernel))))
+        H = fft(kernel)
+        # TODO: store this as well.
+        partial_current = np.real(
+            ifft(fft(v_sig) * np.conj(H) / (H * np.conj(H) + (1 / snr) ** 2))
```

Comment:
Right, so I have no idea what goes on here. A comment, possibly just with a reference would be nice.